### PR TITLE
Dependencies update

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

### Dependencies count
before 203, after 155 = **-48**

### Dependencies updates
 - React scripts 4.0.3 -> 5.0.1
 - Craco 5.7.0 -> 7.0.0

 - Jest 25.2.1 -> 27.5.2
 - Cypress 7.7.0 -> 12.3.0
 - Es-lint 7.11.0 -> 8.32.0

 - React 17.0.1 -> 18.2.0
 - React router 5.3.3 -> 6.6.2
 - Typescript 4.4.3 -> 4.9.4
 
 - @use-gesture/react 6.0.14 -> 10.2.23
 - @react-spring/web 8.0.27 -> 9.6.1

### Main changes
1. (error) => (error: any)
2. import { smth } from 'file.json' -> import data from 'file.json'
3. **React router migration** (useLocation, useNavigate, etc.)

### Side effects
1. Aligned dependencies / devDependencies
2. Cleaned eslint config. A new version of CRA contains everithing we need.
3. Added fix-ethereumjs-bundle.js instead of @ethereumjs patches
5. `pages/MigrateV2/MigrateV2Pair.tsx`, `pages/MigrateV2/index.tsx`, `src/pages/App.tsx` - are removed, just to skip router update changes
6. Removed outdated polyfills

# To Test
1. Theme changes from URL query
2. Expert mode changes from URL query
3. Links, routing, active state of links
4. **Full regression of the whole app**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,95 +1,3 @@
...
-    "@typescript-eslint/explicit-module-boundary-types": "off",
-    "react/react-in-jsx-scope": "off",
-    "object-shorthand": ["error", "always"],
-    "no-restricted-imports": [
-      "error",
-      {
-        "paths": [
-          {
-            "name": "ethers",
-            "message": "Please import from '@ethersproject/module' directly to support tree-shaking."
```
Comment (anxolin) [inline .eslintrc.json:68]: we didn't add these, but  might be good, is double-checking we use the specific modules instead of the `ethers` package

File: .eslintrc.json [hunk 2]
```diff
@@ -1,95 +1,3 @@
...
-          },
-          {
-            "name": "styled-components",
-            "message": "Please import from styled-components/macro."
-          },
-          {
-            "name": "@lingui/macro",
-            "importNames": ["t"],
-            "message": "Please use <Trans> instead of t."
```
Comment (anxolin) [inline .eslintrc.json:77]: aren't we interested in this checks ay more?

File: craco.config.js [hunk 3]
```diff
@@ -60,9 +49,25 @@ module.exports = {
     // https://webpack.js.org/configuration
     configure: (webpackConfig) => ({
       ...webpackConfig,
+      // https://github.com/facebook/create-react-app/discussions/11767#discussioncomment-2421668
```
Comment (anxolin) [inline craco.config.js:52]: I think we might be able in the future to even kill `craco`

File: craco.config.js [hunk 4]
```diff
@@ -60,9 +49,25 @@ module.exports = {
...
+        },
+      ],
       resolve: {
         ...webpackConfig.resolve,
         modules: [path.resolve(__dirname, 'src/custom'), ...webpackConfig.resolve.modules],
+        fallback: {
+          http: require.resolve('stream-http'),
+          https: require.resolve('https-browserify'),
+          crypto: require.resolve('crypto-browserify'),
+        },
```
Comment (anxolin) [inline craco.config.js:70]: Why was this needed?

File: package.json [hunk 5]
```diff
@@ -1,157 +1,21 @@
...
-  },
+  "description": "CoW Swap",
+  "main": "index.js",
+  "author": "",
+  "license": "ISC",
   "scripts": {
+    "start": "craco start",
+    "build": "yarn i18n:compile && craco build && yarn writeVersion",
+    "test": "NODE_PATH=src/custom craco test --env=jsdom",
     "start:default": "craco start",
```
Comment (anxolin) [inline package.json:12]: do we need it if we have the `start` already?

File: src/components/NavigationTabs/index.tsx [hunk 6]
```diff
@@ -71,10 +71,10 @@ const StyledArrowLeft = styled(ArrowLeft)`
 export function SwapPoolTabs({ active }: { active: 'swap' | 'pool' }) {
   return (
     <Tabs style={{ marginBottom: '20px', display: 'none', padding: '1rem 1rem 0 1rem' }}>
-      <StyledNavLink id={`swap-nav-link`} to={'/swap'} isActive={() => active === 'swap'}>
+      <StyledNavLink id={`swap-nav-link`} to={'/swap'} className={({ isActive }) => (isActive ? 'ACTIVE' : undefined)}>
         <Trans>Swap</Trans>
       </StyledNavLink>
-      <StyledNavLink id={`pool-nav-link`} to={'/pool'} isActive={() => active === 'pool'}>
+      <StyledNavLink id={`pool-nav-link`} to={'/pool'} className={({ isActive }) => (isActive ? 'ACTIVE' : undefined)}>
```
Comment (anxolin) [inline src/components/NavigationTabs/index.tsx:77]: feel a bit strange. 
shoudn't `StyledNavLink` receive a `isActive: bool` as param and conditionally add the CSS? 

just curious, no need to fix all the Uniswap code. This PR scope is already big

File: src/cow-react/modules/application/containers/App/RoutesApp.tsx [hunk 7]
```diff
@@ -41,43 +45,154 @@ function createRedirectExternal(url: string) {
...
+        </Route>
+        <Route path="claim" element={<Navigate to={RoutesEnum.ACCOUNT} />} />
+        <Route path="profile" element={<Navigate to={RoutesEnum.ACCOUNT} />} />
+
+        {/*Swap*/}
+        <Route path={RoutesEnum.SWAP} element={<NewSwapPage />} />
+        <Route path={RoutesEnum.SEND} element={<RedirectPathToSwapOnly />} />
+
+        {/*Limit orders*/}
+        <Route
```
Comment (anxolin) [inline src/cow-react/modules/application/containers/App/RoutesApp.tsx:63]: There's a bit of repetition here, should we just do a loop?

**Review by anxolin — APPROVED** [review]
Great job!

File: .eslintrc.json (hunk 2 above)
Comment (shoom3301) [reply .eslintrc.json:77]: I see, that we use `t` a lot, so, I think the rule is not relevant

File: craco.config.js (hunk 3 above)
Comment (shoom3301) [reply craco.config.js:52]: Yes, but we still need it for:
```
modules: [path.resolve(__dirname, 'src/custom'), ...webpackConfig.resolve.modules],
```

Without that code it doesn't work

File: craco.config.js (hunk 4 above)
Comment (shoom3301) [reply craco.config.js:70]: Good question, thanks!
```
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "https": require.resolve("https-browserify") }'
	- install 'https-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "https": false }
```

File: package.json (hunk 5 above)
Comment (shoom3301) [reply package.json:12]: Yep, we don't need it, thanks!

File: src/components/NavigationTabs/index.tsx (hunk 6 above)
Comment (shoom3301) [reply src/components/NavigationTabs/index.tsx:77]: image

Internally it uses className.
React router v6 doesn't support `isActive` prop anymore.
https://reactrouter.com/en/main/components/nav-link

Comment (shoom3301) [reply src/components/NavigationTabs/index.tsx:77]: Anyway, we don't use this file, must be deleted soon as a part of Unswap code.

File: src/cow-react/modules/application/containers/App/RoutesApp.tsx (hunk 7 above)
Comment (shoom3301) [reply src/cow-react/modules/application/containers/App/RoutesApp.tsx:63]: image

Thanks! Looks fancy :)

File: .eslintrc.json (hunk 2 above)
Comment (shoom3301) [reply .eslintrc.json:77]: image

I found, that we use it a lot and just disable this rule, so I think it's not relevant

File: .eslintrc.json (hunk 1 above)
Comment (shoom3301) [reply .eslintrc.json:68]: Looks relevant, got it back

**Comment by shoom3301** [comment]
@elena-zh thank you! Fixed

**Comment by elena-zh** [comment]
Hey @shoom3301 , the 2nd case is still not fixed: 
image

Plus now 2 FAQ sections are highlighted at a time: 
image

**Comment by shoom3301** [comment]
@elena-zh thanks!
Fixed the FAQ side menu links.
{confirmWord} - I can't find the root of the problem :( I remember, @nenadV91 fixed smth similar, will ask him for help

**Comment by elena-zh** [comment]
Hey @nenadV91 , could you please take a look into the [1st issue](https://github.com/cowprotocol/cowswap/pull/1878#issuecomment-1396958561)? It is reproducible only in 1 place where a user need to type in 'Confirm' when switching ON expert mode.

**Comment by elena-zh** [comment]
All the rest changes LGTM!

File: src/custom/serviceWorker/index.ts [hunk 8]
```diff
@@ -6,11 +6,13 @@ import { precacheAndRoute } from 'workbox-precaching'
 import { PrecacheEntry } from 'workbox-precaching/_types'
 import { registerRoute, Route } from 'workbox-routing'
 import { CacheFirst } from 'workbox-strategies'
-import { version as WEB_VERSION } from '@src/../package.json'
+import pckg from '@src/../package.json'
```
Comment (alfetopito) [inline src/custom/serviceWorker/index.ts:9]: Nitpick: `pkg`

**Comment by elena-zh** [comment]
Hey @shoom3301 , lingui issue was addressed and LGTM here. 
But I have found 1 more issue that I have noticed in the PR: Approve token button is not auto-updated on the Tokens page when TX is mined: 
not updated

File: .eslintrc.json [hunk 9]
```diff
@@ -70,26 +13,20 @@
...
-          }
-        ],
-        "patterns": [
-          {
-            "group": ["**/dist"],
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply .eslintrc.json]: Thanks! Added it

File: src/cow-react/abis/types/GPv2Settlement.ts [hunk 10]
```diff
@@ -0,0 +1,153 @@
+/* Autogenerated file. Do not edit manually. */
+/* tslint:disable */
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply src/cow-react/abis/types/GPv2Settlement.ts:2]: I don't know what exactly changed, but now the project doesn't build without these files:
image

**Comment by shoom3301** [comment]
> Hey @shoom3301 , lingui issue was addressed and LGTM here. But I have found 1 more issue that I have noticed in the PR: Approve token button is not auto-updated on the Tokens page when TX is mined: not updated

@elena-zh hi!
I can't reproduce it now, could you recheck it pls?

**Comment by elena-zh** [comment]
> > Hey @shoom3301 , lingui issue was addressed and LGTM here. But I have found 1 more issue that I have noticed in the PR: Approve token button is not auto-updated on the Tokens page when TX is mined: not updated
> 
> @elena-zh hi! I can't reproduce it now, could you recheck it pls?

Hey @shoom3301 , I still can reproduce the issue. See the [video](https://www.loom.com/share/be1893ed38344d5191008130ef3e3484)

**Review by elena-zh — APPROVED** [review]
Hey @shoom3301 , I have retested the changes once again and.. I can't reproduce the issue again. 
So I'm approving the current changes and will keep my eye on it. I will create a separate task for this once this happens once again. 
Thanks

**Comment by alfetopito** [comment]
@SocketSecurity ignore keccak@3.0.3
    @SocketSecurity ignore secp256k1@4.0.3
    @SocketSecurity ignore cypress@12.4.1
    @SocketSecurity ignore core-js@3.27.1
    @SocketSecurity ignore core-js-pure@3.27.1

**Comment by alfetopito** [comment]
@SocketSecurity ignore styled-components@5.3.6

**Comment by alfetopito** [comment]
@SocketSecurity ignore jest@27.5.1
@SocketSecurity ignore jest-cli@27.5.1
